### ISSUE TITLE
New user E-mail sent out on accessibility or/and performance test 

### ISSUE THREAD

**Issue by ipajen** [10up/plugin-check#424]
### Description
When checking accessibility or/and performance test a email is sent out with admin credentials for new site. (System failed to delivery e-mail)

https://wordpress.slack.com/archives/C1LBM36LC/p1707869533873209

### Environment
- WordPress: 6.5-beta1
- PHP: 8.3.2
- Server: LiteSpeed
- Database: mysqli (Server: 10.6.16-MariaDB / Client: mysqlnd 8.3.2)
- Browser: Safari 17.3.1 (Mobile) (iPhone)
- Theme: GeneratePress 3.4.0
- MU-Plugins: 
  * block-automation-by-installatron.php 
  * Health Check Troubleshooting Mode 1.9.0
  * ManageWP - Worker Loader 1.0.0
- Plugins: 
  * Core Rollback 1.3.5
  * Enable jQuery Migrate Helper 1.4.0
  * Health Check & Troubleshooting 1.7.0
  * Limit Login Attempts Reloaded 2.26.4
  * Plugin Check 1.0.0
  * Plugin Compatibility Checker 3.0.1
  * Query Monitor 3.15.0
  * WordPress Beta Tester 3.5.5
  * WP-Cron Status Checker 1.2.5
  * WP Crontrol 1.16.1
  * WP Mail Logging 1.12.0
  * WP SMS 6.5.4
  * WPVulnerability 3.1.1

### Steps to Reproduce
1.  Install and activate PPC version 1.0.0 
2.  Tick accessibility or/and performance test and run check on any plugin
3. 🐞 Bug occurs. 

### Expected Results
1.  ✅  Disable the e-mail sending. Or does a new user need to be created for the test? 

### Actual Results
1.  ❌ System Mail Delivery e-mail sent out on every check - Mail delivery failed: returning message to sender 
 demo@plugincheck.test
   Unrouteable address
   
With a new user (administrator) and password for “ New WordPress Site”

**Comment by dd32** [comment]
Code in question:

https://github.com/WordPress/plugin-check/blob/17a85acefcd157a92d1a798850638d01660c7194/includes/Checker/Runtime_Environment_Setup.php#L51-L56

and the core function that is responsible:
https://github.com/WordPress/wordpress-develop/blob/8ec4d9dfc7e2043485d83d4af9e505ec1cc21470/src/wp-admin/includes/upgrade.php#L535-L616

Filtering that filter and returning a falsey value would work.. but filtering `wp_mail()` as a whole is probably a better route.
